lib/kube/proxy/resource_filters_test.go

```diff
@@ -0,0 +1,256 @@
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *appsv1.DaemonSetList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *appsv1.StatefulSetList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *authv1.RoleBindingList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *batchv1.CronJobList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *batchv1.JobList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *corev1.PodList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *corev1.ConfigMapList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *corev1.ServiceAccountList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *appsv1.ReplicaSetList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *corev1.ServiceList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *corev1.PersistentVolumeClaimList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *authv1.RoleList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *networkingv1.IngressList:
+					resources = collectResourcesFromResponse(arrayToPointerArray(o.Items))
+				case *metav1.Table:
+					for i := range o.Rows {
+						row := &(o.Rows[i])
+						if row.Object.Object == nil {
+							var err error
+							// decode only if row.Object.Object was not decoded before.
+							row.Object.Object, err = decodeAndSetGVK(decoder, row.Object.Raw)
+							require.NoError(t, err)
+						}
+
+						resource, err := getKubeResourcePartialMetadataObject(r, row.Object.Object)
+						require.NoError(t, err)
+						resources = append(resources, resource.Namespace+"/"+resource.Name)
+					}
+				default:
+					t.Errorf("filterBuffer() = %v (%T)", obj, obj)
+				}
+				sort.Strings(resources)
+				sort.Strings(tt.want)
+				require.Equal(t, tt.want, resources)
+			})
+		}
+	}
+}
+
+func collectResourcesFromResponse[T kubeObjectInterface](originalList []T) []string {
+	resources := []string{}
+	for _, resource := range originalList {
+		resources = append(resources, resource.GetNamespace()+"/"+resource.GetName())
+	}
+	return resources
+}
+
+func newMemoryResponseWriter(t *testing.T, payload []byte, contentEncoding string) (*responsewriters.MemoryResponseWriter, decompressionFunc) {
+	buf := responsewriters.NewMemoryResponseWriter()
+	buf.Header().Set(contentEncodingHeader, contentEncoding)
+	buf.Header().Set(responsewriters.ContentTypeHeader, responsewriters.DefaultContentType)
+
+	switch contentEncoding {
+	case "gzip":
+		w, err := gzip.NewWriterLevel(buf, defaultGzipContentEncodingLevel)
+		require.NoError(t, err)
+		defer w.Close()
```

Comment:
Yes, `w`is the gzip writer that outputs to `buf`. After `_, err = w.Write(payload)` we can flush and close it because the compressed data is already in `buf` to be used later.